pkg/strvals/parser.go

```diff
@@ -375,8 +380,12 @@ func typedVal(v []rune, st bool) interface{} {
 		return false
 	}
 
-	// If this value does not start with zero, and not returnString, try parsing it to an int
-	if !st && len(val) != 0 && val[0] != '0' {
+	if strings.EqualFold(val, "null") {
+		return nil
+	}
+
```

Comment:
In the current v2 there is also...

https://github.com/helm/helm/blob/0ca95341c19fcaf9cecdf50caef1693568835a84/pkg/strvals/parser.go#L396-L398

It's between these spaces. Should that be here, too.

Reply:
yes, good catch. That was from #5151. I picked up the null fix from #3912.

Reply:
fix applied.